core/utils/dist_checker.py

```
@@ -154,6 +154,58 @@ def _extract_dist(zip_bytes: bytes, dist_dir: Path) -> None:
+    """Download and verify the WebUI archive for a release without applying it."""
+    direct_url = f"https://github.com/{REPO_OWNER}/{REPO_NAME}/releases/download/{version}/{ASSET_NAME}"
+    ranked_urls = await pick_fastest_source(direct_url, proxy)
+    if not ranked_urls:
+        raise RuntimeError("All WebUI download sources failed")
+
+    expected_sha = await _fetch_asset_digest(version, proxy)
+    for index, download_url in enumerate(ranked_urls):
+        logger.info(f"Downloading {ASSET_NAME} (source {index + 1}/{len(ranked_urls)}) ...")
+        try:
+            content = await get_file_content(download_url, proxy=proxy, timeout=30.0)
+        except Exception as exc:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)